v1/plugin/plugin.go

```diff
@@ -57,6 +58,47 @@ type Publisher interface {
 	Publish([]Metric, Config) error
 }
 
+// StreamCollector is a Collector that can send back metrics on it's own
+// defined interval (within configurable limits). These limits are set by the
+// SetMaxBuffer and SetMaxCollectionDuration funcs.
+type StreamCollector interface {
+	Plugin
+
+	// StreamMetrics allows the plugin to send/receive metrics on a channel
+	// Arguments are (in order):
+	//
+	// A channel for metrics into the plugin from Snap -- which
+	// are the metric types snap is requesting the plugin to collect.
+	//
+	// A channel for metrics from the plugin to Snap -- the actual
+	// collected metrics from the plugin.
+	//
+	// A channel for error strings that the library will report to snap
+	// as task errors.
+	StreamMetrics(chan []Metric, chan []Metric, chan string) error
+	// SetMaxbuffer sets the maximum number of metrics the plugin should buffer
+	// before sending metrics.
+	SetMaxBuffer(int64)
```

Comment:
@IRCody as we discussed offline let's move SetMaxBuffer and SetMaxCollectDuration out of the path of the plugin author or stated another way they will be implemented by the lib (in the proxy).  We'll follow-up with a separate PR that adds the impl for these two options so for now it should just be empty.